src/bipart.cpp

```diff
@@ -346,6 +372,86 @@ namespace libsemigroups {
       return result;
     }
 
+    void uniform_random(Bipartition& x) {
+      if (x.degree() == 0) {
+        return;
+      }
+      long double const N = 2 * x.degree();
+
+      static std::random_device rd;
+      static std::mt19937       mt(rd());
+
+      long double const c = M_E * bell_number(N);
+      if (std::isnan(c)) {
+        LIBSEMIGROUPS_EXCEPTION(
+            "the degree ({}) of the argument <x> (a bipartition) is too large, "
+            "please use random(x) instead",
+            N / 2);
+      }
+      std::discrete_distribution<uint64_t> g(
+          4 * N, 0, 4 * N, [&N, &c](long double m) -> long double {
+            long double num = std::pow(m, N);
+            long double den = c * factorial(m);
+            if (std::isnan(num) || std::isnan(den) || std::isnan(num / den)) {
+              LIBSEMIGROUPS_EXCEPTION("the degree ({}) of the argument <x> (a "
+                                      "bipartition) is too large, "
+                                      "please use random(x) instead",
+                                      N / 2);
+            }
+            return num / den;
+          });
+
+      auto                                    M = g(mt) - 1;
```

Comment:
I think this is sometimes causing an underflow error. If I generate enough random bipartitions of degree 1, eventually `g(mt)` will return `0`, so `M` underflows